### PR TITLE
wsd: limit memToFreeKb in Admin::triggerMemoryCleanup()

### PR DESCRIPTION
**Pull request by vmiklos**
(cherry picked from commit 034b37a4a903ffa879f0c27007a3d42b8ecab184)

Signed-off-by: Miklos Vajna <vmiklos@collabora.com>
Change-Id: Idfe4ee00be9f805f5e12c7ece4f890aa2b9d08ed

### CODE REVIEW

**Comment by vmiklos** [comment]
@Ashod can you review this, please? The review is just for the backport, Michaal already reviewed this for master at #5384. Thanks!

File: wsd/Admin.cpp [hunk 1]
```diff
@@ -735,6 +735,12 @@ void Admin::notifyForkit()
     COOLWSD::sendMessageToForKit(oss.str());
 }
 
+/// Similar to std::clamp(), old libstdc++ doesn't have it.
+template <typename T>
+T clamp(const T& n, const T& lower, const T& upper) {
+  return std::max(lower, std::min(n, upper));
```
Comment (Ashod) [inline wsd/Admin.cpp:741]: Needs clang-format, but otherwise good.

File: wsd/Admin.cpp [hunk 2]
```diff
@@ -750,7 +756,7 @@ void Admin::triggerMemoryCleanup(const size_t totalMem)
             memLimit << "% (" << static_cast<size_t>(_totalSysMemKb * memLimit / 100.) << " KB).");
 
     const double memToFreePercentage = (totalMem / static_cast<double>(_totalSysMemKb)) - memLimit / 100.;
-    int memToFreeKb = static_cast<int>(memToFreePercentage > 0.0 ? memToFreePercentage * _totalSysMemKb : 0);
+    int memToFreeKb = clamp<double>(memToFreePercentage > 0.0 ? memToFreePercentage * _totalSysMemKb : 0, 0, std::numeric_limits<int>::max());
```
Comment (Ashod) [inline wsd/Admin.cpp:759]: If we are clamping, then we don't need the conditional. Should be more readable that way.

**Review by Ashod — APPROVED** [review]
Thanks @vmiklos! My comments are generic and wouldn't apply to this backport. (Although best to have them in master.)

File: wsd/Admin.cpp (hunk 1 above)
Comment (vmiklos) [reply wsd/Admin.cpp:741]: Done, see https://github.com/CollaboraOnline/online/pull/5472.

File: wsd/Admin.cpp (hunk 2 above)
Comment (vmiklos) [reply wsd/Admin.cpp:759]: Done, see https://github.com/CollaboraOnline/online/pull/5472.

**Pull request merged**